internal/email/email.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package email
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"html/template"
+	"net/smtp"
+
+	"github.com/mainflux/mainflux/logger"
+)
+
+var (
+	// ErrMissingEmailTmpl missing email template file
+	ErrMissingEmailTmpl = errors.New("Missing email template file")
+)
+
+type emailTemplate struct {
+	To      []string
+	From    string
+	Subject string
+	Header  string
+	Content string
+	Footer  string
+}
+
+// Config email agent configuration.
+type Config struct {
+	Driver      string
+	Host        string
+	Port        string
+	Username    string
+	Password    string
+	FromAddress string
+	FromName    string
+	Template    string
+}
+
+// Agent for mailing
+type Agent struct {
+	conf *Config
+	auth smtp.Auth
+	addr string
+	log  logger.Logger
+	tmpl *template.Template
+}
+
+// New creates new email agent
+func New(c *Config, t *template.Template) *Agent {
+	a := &Agent{}
+	a.conf = c
+
+	// Set up authentication information
+	a.auth = smtp.PlainAuth("", c.Username, c.Password, c.Host)
+	a.addr = fmt.Sprintf("%s:%s", c.Host, c.Port)
+	if t != nil {
+		a.tmpl = t
+		return a
+	}
+
+	tmpl, _ := template.ParseFiles(c.Template)
```

Comment:
Don't ignore error.

Reply:
this is new method  it shuldnt retur error and I cant log the error and anyways if thereis an error tmpl will be nil so it will be reported when trying to send an email

Reply:
OK, but you will wait for that call in order to fail. IMHO if a template is invalid, the whole service should fail with an error.

Reply:
what about situation when service restarts and template somehow was deleted, whole service will be unavailable cause 10% of functionality is not available?